### PR TITLE
Fix the StateBinary.get32 parser

### PR DESCRIPTION
**Pull request by erikd**
Found when round-tripping packets captured with Wireshark.

### CODE REVIEW

File: Network/DNS/StateBinary.hs [hunk 1]
```diff
@@ -125,7 +125,7 @@ get16 = ST.lift getWord16be <* addPosition 2
     getWord16be = do
         a <- word8'
         b <- word8'
-        return $ a * 256 + b
+        return $ a * 0x100 + b
```
Comment (kazu-yamamoto) [inline Network/DNS/StateBinary.hs]: Is this a just style change?

File: Network/DNS/StateBinary.hs [hunk 2]
```diff
@@ -136,7 +136,7 @@ get32 = ST.lift getWord32be <* addPosition 4
         b <- word8'
         c <- word8'
         d <- word8'
-        return $ a * 1677721 + b * 65536 + c * 256 + d
+        return $ a * 0x1000000 + b * 0x10000 + c * 0x100 + d
```
Comment (kazu-yamamoto) [inline Network/DNS/StateBinary.hs]: Does this mean that I made a mistake on copying `16777216`?

File: Network/DNS/StateBinary.hs (hunk 1 above)
Comment (erikd) [inline Network/DNS/StateBinary.hs]: Yes, most coders know that `256` is `0x100 but few coders can tell the decimal value of `0x1000000`. Changing this one so its in the same form as the others below.

File: Network/DNS/StateBinary.hs (hunk 2 above)
Comment (erikd) [inline Network/DNS/StateBinary.hs]: I think you know the answer to that :).

Comment (kazu-yamamoto) [inline Network/DNS/StateBinary.hs]: Thank you for finding this. :-)

**Comment by kazu-yamamoto** [comment]
Merged.
Thank you for your contributions!

File: Network/DNS/StateBinary.hs (hunk 2 above)
Comment (erikd) [inline Network/DNS/StateBinary.hs]: Thank you for providing this library. *Soooo* much easier to extend this than starting from scratch.

**Pull request merged**